app/controllers/batch_update_tools_controller.rb

```diff
@@ -9,26 +10,58 @@ def show
                 else
                   []
                 end
+
+    classes_file = File.open("app/assets/data/classes.json")
+    classes_data = classes_file.read
+    @classes = JSON.parse(classes_data)
   end
 
   def update
     @field = Field.signoffs
     @values = @field.allowed_values
+    classes_file = File.open("app/assets/data/classes.json")
+    classes_data = classes_file.read
+    @classes = JSON.parse(classes_data)
+    puts "found the following classes"
+    puts @classes
 
     @contacts = params["contacts"].uniq
     @contacts.each do |uid|
       user = User.find_by uid: uid
       values = user.signoff_values
-      values << params["field_value"]
-
+    # check if the user submitted a class or a single sign off
+      if params["field_value"].match(/\[class\]/)
+        @classes.each do |aClass|
+          if aClass["class_name"] == params["field_value"]
+            #found the right class in our data. Add all of its signoffs to values, so they'll get sent to WA
+            #It looks like WA can handle duplicate signoffs just fine, so no need to dedup the array
+            puts "going with " + aClass["class_name"]
+            temp_values = []
+            aClass["signoffs_granted"].each do |signoff_name|
+              temp_values << FieldAllowedValue.find_by(label: signoff_name).uid
+            end            
+          values.concat(temp_values)
+          end
+          puts "found class " + aClass["class_name"]
+          puts "with signoffs " 
+          puts aClass["signoffs_granted"]
+          puts "mapping to "
+          puts temp_values
+          puts "for a final submission of"
+          puts values
```

Comment:
Would be good to drop the `puts`'s

Reply:
right right. forgot to remove my debug outputs